src/pages/index.js

```diff
@@ -33,53 +48,120 @@ const cardSection = new Section({
   }
 }, ".cards");
 const popupWithImage = new PopupWithImage('.modal_type_image');
-const api = new Api({
-  baseUrl: "https://around-api.en.tripleten-services.com/v1",
-  headers: {
-    authorization: "32400e4e-bdb6-4554-9f2b-339b0c55f4e6",
-    "Content-Type": "application/json"
-  }
-});
+
 
 
 // define functions
 function renderProfileDetails(){
-  const { name, description } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   nameInput.value = name;
-  jobInput.value = description;
+  jobInput.value = about;
 };
 
 function handleCardAdd(data){
-  const { title, url } = data
-  const cardElement = createCard({title, url})
-  cardSection.addItem(cardElement)
-  cardModal.close()
+  const submitButton = cardAddFormElement.querySelector('.modal__save-button')
+  const text = submitButton.textContent;
+  submitButton.textContent = "Saving..."
+
+  const { name, link } = data
+  api.addCard(name, link).then((data)=>{
+    cardSection.addItem(createCard(data))
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    submitButton.textContent = text;
+    cardModal.close()
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well